lib/reline/unicode.rb

```diff
@@ -160,24 +160,29 @@ def self.split_by_width(str, max_width, encoding = str.encoding)
     width = 0
     rest = str.encode(Encoding::UTF_8)
     in_zero_width = false
+    seq = String.new(encoding: encoding)
     rest.scan(WIDTH_SCANNER) do |gc|
       case
       when gc[NON_PRINTING_START_INDEX]
         in_zero_width = true
+        lines.last << NON_PRINTING_START
       when gc[NON_PRINTING_END_INDEX]
         in_zero_width = false
+        lines.last << NON_PRINTING_END
       when gc[CSI_REGEXP_INDEX]
         lines.last << gc[CSI_REGEXP_INDEX]
+        seq << gc[CSI_REGEXP_INDEX]
       when gc[OSC_REGEXP_INDEX]
         lines.last << gc[OSC_REGEXP_INDEX]
+        seq << gc[OSC_REGEXP_INDEX]
```

Comment:
I think this addition is not covered by tests. Can you add a test case for it?

Reply:
I added 😄